node/src/components/block_synchronizer/block_acquisition.rs

```diff
@@ -762,11 +798,11 @@ impl BlockAcquisitionState {
                     None
                 }
             }
-            BlockAcquisitionState::HaveBlock(block, acquired_signatures, acquired_deploys) => {
+            BlockAcquisitionState::HaveBlock(block, acquired_signatures, acquired_transactions) => {
                 maybe_block_hash = Some(*block.hash());
                 acceptance = acquired_signatures.apply_signature(signature, validator_weights);
                 if !is_historical
-                    && acquired_deploys.needs_deploy().is_none()
+                    && acquired_transactions.needs_transaction() == false
```

Comment:
A nit:
I understand the `== false` vs `!` discussion, but using both approaches in a single `if` statement might be confusing.